### PR TITLE
Small change to tolerate exception info in show-configuration #179

### PR DESCRIPTION
**Pull request by stopthatastronaut**
Handles issue #179

### CODE REVIEW

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 [hunk 1]
```diff
@@ -215,6 +215,31 @@ function Get-TargetResource {
...
+
+function Get-CleanedJson
+{
+    param($jsonstring)
+    $jsonstart = $jsonstring.IndexOf("{")
+    Write-Verbose "Found start of JSON at character $jsonstart"
+    $extractedjson = $jsonstring.Substring($jsonstart, $jsonstring.length - $jsonstart)
+
+    $dumpedstring = $jsonstring.substring(0, $jsonstart-1)
+    Write-Verbose "stripped extra content from JSON configuration string`r`n`r`n$dumpedstring"
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: This feels like it should be `write-warning`

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 [hunk 2]
```diff
@@ -237,7 +262,18 @@ function Import-ServerConfig {
 
     if (Test-OctopusVersionSupportsShowConfiguration) {
         $rawConfig = & $octopusServerExePath show-configuration --format=json-hierarchical --noconsolelogging --console --instance $InstanceName
-        $config = $rawConfig | ConvertFrom-Json
+
+        # handle a specific error where an exception in registry migration finds its way into the json-hierarchical output
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: Can you give specific details about what the error was?

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 [hunk 3]
```diff
@@ -237,7 +262,18 @@ function Import-ServerConfig {
...
+        if(Test-ValidJson $rawConfig) {
+            $config = $rawConfig | ConvertFrom-Json
+        } else {
+            $cleanedUpConfig = Get-CleanedJson $rawConfig
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: Maybe stick a `write-warning "show-configuration returned invalid json. Attempting to clean it up."

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 [hunk 4]
```diff
@@ -237,7 +262,18 @@ function Import-ServerConfig {
 
     if (Test-OctopusVersionSupportsShowConfiguration) {
         $rawConfig = & $octopusServerExePath show-configuration --format=json-hierarchical --noconsolelogging --console --instance $InstanceName
-        $config = $rawConfig | ConvertFrom-Json
+
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: Might be worth sticking a pester test around this

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 (hunk 1 above)
Comment (stopthatastronaut) [reply OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: I changed it from write waring in a fit of self-doubt. Happy to change it back if you are.

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 (hunk 2 above)
Comment (stopthatastronaut) [reply OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: I'll add a reference back to the Issue #179

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 (hunk 3 above)
Comment (stopthatastronaut) [reply OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: sorted

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 [hunk 5]
```diff
@@ -237,7 +262,21 @@ function Import-ServerConfig {
...
+        if(Test-ValidJson $rawConfig) {
+            $config = $rawConfig | ConvertFrom-Json
+        } else {
+            Write-Warning "Invalid json encountered in show-configuration; attempting to clean up."
+            $cleanedUpConfig = Get-CleanedJson $rawConfig
+            if(Test-ValidJson $cleanedUpConfig ) {
+                $config = $cleanedUpConfig | ConvertFrom-Json
+            } else {
+                Write-Error "Attempted to cleanup bad JSON and failed. String we attempted to parse was`r`n`r`n$rawConfig"
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: Could this include a connection string? Or a master key?
Are we masking?

Comment (stopthatastronaut) [reply OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: Removed the bit of potentially leaky code

File: OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1 (hunk 4 above)
Comment (stopthatastronaut) [reply OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1]: Added a couple of simple pester tests.

File: OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1 [hunk 6]
```diff
@@ -131,4 +134,21 @@ Describe "Invoke-OctopusServerCommand" {
...
+
+    It "returns true for known good json" {
+        Test-ValidJson (Get-Content "$SamplePath\octopus.server.exe-output-clean.json" -raw) | Should Be $true
+    }
+}
+
+Describe "Get-CleanedJson" {
+    It "Correctly cleans our expected exception-prepended output" {
+        $clean = Get-CleanedJson (Get-Content "$SamplePath\octopus.server.exe-output-when-json-has-exception-prepended.json" -raw)
+        Test-ValidJson $clean | Should Be $true
```
Comment (matt-richardson) [inline OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1]: Can you also compare the cleaned json to `octopus.server.exe-output-clean.json`?

Comment (stopthatastronaut) [reply OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1]: Probably a damn good plan. I'll add that

**Pull request merged**